jlm/llvm/ir/operators/MemoryStateOperations.cpp

```diff
@@ -279,6 +282,30 @@ LambdaExitMemoryStateMergeOperation::copy() const
   return std::make_unique<LambdaExitMemoryStateMergeOperation>(*this);
 }
 
+rvsdg::Input *
+LambdaExitMemoryStateMergeOperation::MapMemoryNodeIdToInput(
+    const rvsdg::Node & node,
+    const MemoryNodeId memoryNodeId)
+{
+  const auto operation =
+      dynamic_cast<const LambdaExitMemoryStateMergeOperation *>(&node.GetOperation());
```

Comment:
I think maybe Helge was moving away from nodes having operations (only `SimpleNode` having them), but we can handle that when we get there